libethcore/Miner.h

```diff
@@ -241,6 +282,7 @@ class Miner: public Worker
 	std::chrono::high_resolution_clock::time_point workSwitchStart;
 	HwMonitorInfo m_hwmoninfo;
 private:
+	bool     m_wait_for_tstart_temp = false;
```

Comment:
Would make m_wait_for_tstart_temp a std::atomic<bool> as we're querying from different threads.

Reply:
Why it must be accessed from different threads? If that's required make it an `atomic_flag`.

Reply:
Cause each miner lives in it's own thread and the value is getting accessed by Farm which lives on a separate thread.

Reply:
@chfast , @AndreaLanfranchi 
As I'm new in this topic (C++, C11) - do we really need this?
As m_wait_for_tstart_temp is just updated (modified) in `update_temperature()` in Miner.h which is called from the farm and I don't think its a critical section  - do I have to use `atomic_flag` ?

Reply:
Yes, you have to. Otherwise this will be a race condition and Thread Sanitizer will report that as an issue.